### ISSUE TITLE
💁🏻‍♂️ Feedback and suggestions about localization within Firebase

### ISSUE THREAD

**Issue by ScreamZ** [firebase/firebaseui-web#662]
# In 2020, localization is not an option anymore

It's a first-class feature that is required in almost every project.

**I know that firebase is free**, but still… I'm **paying for services due to my usage**. 
I think this would be a great thing for both firebase platform and users to be able to translate this module correctly. You'll reach more developers and we would be even happier to build apps ;)

This is not something very hard I guess when we know the technical level of people like @CodingDoug @jenperson @ToddKerpelman and many others @ firebase (ping @nicolasgarnier).

## Problem with current configuration

- **Who is using UMD module in 2020 ?**

I don't think many people keep using the UMD CDN build, most people build their app using react/vue or even angular… (I'm sorry WordPress users 😅)

Everything gets packaged in something like webpack or rollup… ES6 / CommonJS is a need.

- **Build time configuration is not adaptative**

If you have a multi-localized app, it's not scalable. You can't even simply update the lib to leverage patches.

## Simple Solution

- Let the library close to what it is currently, with English shipped, so the firebase team has **only one language to maintain**.
- Give the opportunity to pass a localized object with keys/values at instantiation, like we have `signInFlow` currently, we could have a `translations` which will be dynamically injected by us, depending on any context.

I would be very happy to help with that, but I think It would be quicker for me to rework all of this with react-js due to the complexity of the code here Ahah 😅 But even though, I can help on some parts (@nicolasgarnier if you need some help on react)

Best wishes for this Christmas,

Andréas

**Comment by barbalex** [comment]
I am developing an app for users in Switzerland, using firebase auth. I can configure email to arrive in German. But the Login pops up in English and that is not adjustable?

Somehow this makes me thing that firebase auth could be the next app that Google will suddenly kill. It certainly does not look like it seems to get any love.

So I guess I will have to use a different service.

**Comment by bojeil-google** [comment]
Hey @barbalex, please follow the instructions on localization as documented in the [README](https://github.com/firebase/firebaseui-web#building-firebaseui). We have supported localization for years. Google does realize the importance of supporting non-English languages.

**Comment by barbalex** [comment]
@bojeil-google  Please tell me how I should do that in a Gatsby app using firebaseui-web-react. There is an issue about this here: https://github.com/firebase/firebaseui-web-react/issues/21.

**Comment by barbalex** [comment]
Plus: This issue would enable using your own wording. For instance: German is not the same as Swiss German. And not every app would want to use the same sentences. A more flexible approach would help a lot.

**Comment by barbalex** [comment]
Please don't understand me wrong: I am not trying to talk firebase auth bad. I am trying to use it instead of auth0. Which was WAY to complicated. So far I have really enjoyed the dev experience with firebase auth. This has been the first hickup.

**Comment by bojeil-google** [comment]
Hey @barbalex, we do support a lot of [languages](https://github.com/firebase/firebaseui-web/blob/master/LANGUAGES.md) but unfortunately we can't support all the languages in the world. You can also add your own customized translation by plugging it [here](https://github.com/firebase/firebaseui-web/tree/master/translations) and then rebuilding with your own translations.

**Comment by barbalex** [comment]
Yeah, but how am I going to integrate the rebuilt version with firebaseui-web-react?

**Comment by bojeil-google** [comment]
I will reach out to colleagues who own the react version to respond to your question. Unfortunately, I am not the react expert.
It may be hard to perceive, but even Firebase has limited resources. We have tons of things we are working on which we think offer great value to our developers and that developers will love. This means we have to prioritize things on our end and shift resources from one place to another. That sometimes also means we can't build everything that gets requested (at least not in the timelines that people expect) and we do get a lot of requests.

**Comment by barbalex** [comment]
They (or is that them?) have contacted you here https://github.com/firebase/firebaseui-web-react/issues/21#issuecomment-371528237 requesting npm-versions of the localizations. That was two years ago and nothing has changed.

This issue proposes another solution that might be even better.

I realize that not everything is possible. Thanks anyway for your feedback.

**Comment by jhuleatt** [comment]
@barbalex for gatsby-specific localization questions, please file a new issue in firebaseui-web-react and @ me so that I can help without cluttering this thread. Let's keep this issue focused on the original feature request to make firebaseui-web itself more configurable.

**Comment by landsman** [comment]
Guys let's do something with this, please. It's a shame.

As I mentioned here: https://github.com/firebase/firebaseui-web-react/issues/21#issuecomment-987801917. The community can help, if you will prepare the battle plan 🙏

**Comment by dackers86** [comment]
Thank you for taking the time to submit this issue.

A new release is planned shortly, which may resolve this request. 

In the meantime, we will close this issue, if this is not resolved in the new version - please feel free to create a new issue.
